src/VTableDumper.cpp

```diff
@@ -0,0 +1,216 @@
+        static std::filesystem::path member_variable_layouts_gen_output_src_path = member_variable_layouts_gen_output_path / "generated_src";
+        static std::filesystem::path member_variable_layouts_gen_function_bodies_path = member_variable_layouts_gen_output_include_path / "FunctionBodies";
+        static std::filesystem::path member_variable_layouts_templates_output_path = "MemberVarLayoutTemplates";
+        static std::filesystem::path virtual_gen_output_path = "GeneratedVirtualImplementations";
+        static std::filesystem::path virtual_gen_output_include_path = virtual_gen_output_path / "generated_include";
+        static std::filesystem::path virtual_gen_function_bodies_path = virtual_gen_output_include_path / "FunctionBodies";
+
+        File::StringType pdb_name = symbols.pdb_file_path.filename().stem();
+
+        if (std::filesystem::exists(vtable_gen_output_include_path))
+        {
+            for (const auto& item : std::filesystem::directory_iterator(vtable_gen_output_include_path))
+            {
+                if (item.is_directory()) { continue; }
+                if (item.path().extension() != STR(".hpp") && item.path().extension() != STR(".cpp")) { continue; }
+
+                File::delete_file(item.path());
+            }
+        }
+
+        if (std::filesystem::exists(vtable_gen_output_function_bodies_path))
+        {
+            for (const auto& item : std::filesystem::directory_iterator(vtable_gen_output_function_bodies_path))
+            {
+                if (item.is_directory()) { continue; }
+                if (item.path().extension() != STR(".cpp")) { continue; }
+
+                File::delete_file(item.path());
+            }
+        }
+
+        if (std::filesystem::exists(vtable_gen_output_src_path))
+        {
+            for (const auto& item : std::filesystem::directory_iterator(vtable_gen_output_src_path))
+            {
+                if (item.is_directory()) { continue; }
+                if (item.path().extension() != STR(".cpp")) { continue; }
+
+                File::delete_file(item.path());
+            }
+        }
+
+        for (const auto& [class_name, class_entry] : type_container.get_class_entries())
+        {
+            Output::send(STR("Generating file '{}_VTableOffsets_{}_FunctionBody.cpp'\n"), pdb_name, class_entry.class_name_clean);
+            Output::Targets<Output::NewFileDevice> function_body_dumper;
+            auto& function_body_file_device = function_body_dumper.get_device<Output::NewFileDevice>();
+            function_body_file_device.set_file_name_and_path(vtable_gen_output_function_bodies_path / std::format(STR("{}_VTableOffsets_{}_FunctionBody.cpp"), pdb_name, class_name));
+            function_body_file_device.set_formatter([](File::StringViewType string) {
+                return File::StringType{ string };
+                });
```

Comment:
```suggestion
            });
```
Poor indentation, not caused by mixed tabs & spaces, probably IDE giving bad suggestions.